docs/user_directory.md

```diff
@@ -1,49 +1,133 @@
-User Directory API Implementation
-=================================
+# User Directory API Implementation
 
-The user directory is currently maintained based on the 'visible' users
-on this particular server - i.e. ones which your account shares a room with, or
-who are present in a publicly viewable room present on the server.
+The user directory is maintained based on the 'visible' users of a homeserver -
```

Comment:
"visible users of a homeserver" sounds a bit like we control these users, even the remote ones. Maybe:

```suggestion
The user directory is maintained based on users that are 'visible' to the homeserver -
```